waspc/data/Generator/templates/react-app/plugins/validateEnv.ts

```diff
@@ -0,0 +1,14 @@
+import { type Plugin, loadEnv } from "vite";
+
+import { ensureEnvSchema } from 'wasp/env'
+import { clientEnvSchema } from 'wasp/client/env/schema'
+
+export function validateEnv(): Plugin {
+  return {
+    name: 'wasp-validate-env',
+    config: (config) => {
+      const env = loadEnv(config.mode, process.cwd(), config.envPrefix)
+      ensureEnvSchema(env, clientEnvSchema)
```

Comment:
This plugin loads env vars and uses Zod to validate the them using the client env vars schema.